app/src/test/java/com/kks/nimblesurveyjetpackcompose/repo/survey/SurveyRepoImplTest.kt

```diff
@@ -0,0 +1,154 @@
+package com.kks.nimblesurveyjetpackcompose.repo.survey
+
+import com.google.common.truth.Truth.assertThat
+import com.kks.nimblesurveyjetpackcompose.attributeAnswerResponse
+import com.kks.nimblesurveyjetpackcompose.attributeQuestionResponse
+import com.kks.nimblesurveyjetpackcompose.includedAnswerResponse
+import com.kks.nimblesurveyjetpackcompose.includedQuestionResponse
+import com.kks.nimblesurveyjetpackcompose.model.QuestionDisplayType
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
+import com.kks.nimblesurveyjetpackcompose.model.request.SubmitSurveyRequest
+import com.kks.nimblesurveyjetpackcompose.model.response.BaseResponse
+import com.kks.nimblesurveyjetpackcompose.model.response.SurveyIncludedRelationshipsResponse
+import com.kks.nimblesurveyjetpackcompose.model.toSurveyQuestionRequest
+import com.kks.nimblesurveyjetpackcompose.network.Api
+import com.kks.nimblesurveyjetpackcompose.surveyAnswer
+import com.kks.nimblesurveyjetpackcompose.surveyQuestion
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.last
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class SurveyRepoImplTest {
+
+    val api: Api = mockk()
+    lateinit var sut: SurveyRepoImpl
+
+    @Before
+    fun setup() {
+        sut = SurveyRepoImpl(api)
+    }
+
+    @Test
+    fun `When getSurveyDetail and return success`() = runTest {
+        coEvery { api.getSurveyDetail(any()) } returns BaseResponse(
+            included = listOf(
+                includedQuestionResponse,
+                includedAnswerResponse
+            )
+        )
+
+        val expected = sut.getSurveyDetails("0").last()
+
+        assertThat(expected).isInstanceOf(ResourceState.Success::class.java)
+    }
+
+    @Test
+    fun `When getSurveyDetail and return success, question list is in ascending order`() = runTest {
+        coEvery { api.getSurveyDetail(any()) } returns BaseResponse(
+            included = listOf(
+                includedQuestionResponse.copy(
+                    attributes = attributeQuestionResponse.copy(displayOrder = 1),
+                    relationships = SurveyIncludedRelationshipsResponse()
+                ),
+                includedQuestionResponse,
+                includedAnswerResponse
+            )
+        )
+
+        val expected = sut.getSurveyDetails("0").last()
+
+        assertThat(expected).isInstanceOf(ResourceState.Success::class.java)
```

Comment:
For example, this could be 
```suggestion
        assertThat(expected, `is`(instanceOf(ResourceState.Success::class.java)))
```

Reply:
Yes, it does. Google truth is also built on the existing library and not an entirely new library. But it is more readable.